src/api/app/controllers/webui/project_controller.rb

```diff
@@ -698,12 +698,16 @@ def add_maintained_project_dialog
 
   def add_maintained_project
     authorize @project, :update?
-    @project.maintained_projects.new(project: @maintained_project)
-    if @project.save
+
+    @maintained_project = params[:maintained_project]
+    maintained_project = Project.where(name: @maintained_project)
+    if maintained_project.exists?
+      @project.maintained_projects.create!(project: maintained_project.first)
```

Comment:
If `create` fails it raises a `RecordInvalid` error and the reason why you are changing this function is to avoid that the user see an error. Although currently `create` can not fail because there are not validations, I would write it like:

``` Ruby
project = Project.find_by(name: params[:maintained_project])
maintained_project = @project.maintained_projects.create(project: maintained_project) if project
if maintained_project && maintained_project.valid?
  @project.store
  redirect_to({action: 'maintained_projects', project: @project}, notice: "Added #{params[:maintained_project]} to maintenance")
```

That way there is no chance that an exception is raised. :wink: